Neos.ContentRepository.Core/Classes/Projection/ContentGraph/ContentSubgraphInterface.php

```diff
@@ -57,107 +47,94 @@
-        Filter\FindChildNodesFilter $filter
-    ): Nodes;
-
-    public function findReferencedNodes(
-        NodeAggregateId $nodeAggregateId,
-        Filter\FindReferencedNodesFilter $filter
-    ): References;
-
-    public function findReferencingNodes(
-        NodeAggregateId $nodeAggregateId,
-        Filter\FindReferencingNodesFilter $filter
-    ): References;
+    public function findNodeById(NodeAggregateId $nodeAggregateId): ?Node;
 
     /**
-     * @param NodeAggregateId $nodeAggregateId
-     * @return Node|null
+     * Find direct child nodes of the specified parent node
      */
-    public function findNodeById(NodeAggregateId $nodeAggregateId): ?Node;
+    public function findChildNodes(NodeAggregateId $parentNodeAggregateId, Filter\FindChildNodesFilter $filter): Nodes;
 
     /**
-     * @param NodeAggregateId $childNodeAggregateId
-     * @return Node|null
+     * Find the direct parent of a node specified by its aggregate id
+     *
+     * @return Node|null the node or NULL if the specified node is the root node or is inaccessible
      */
     public function findParentNode(NodeAggregateId $childNodeAggregateId): ?Node;
 
     /**
-     * @param NodePath $path
-     * @param NodeAggregateId $startingNodeAggregateId
-     * @return Node|null
+     * Find all nodes that are positioned _behind_ the specified sibling and match the specified $filter
      */
-    public function findNodeByPath(
-        NodePath $path,
-        NodeAggregateId $startingNodeAggregateId
-    ): ?Node;
+    public function findSucceedingSiblingNodes(NodeAggregateId $siblingNodeAggregateId, Filter\FindSucceedingSiblingNodesFilter $filter): Nodes;
 
     /**
-     * @param NodeAggregateId $parentNodeAggregateId
-     * @param NodeName $edgeName
-     * @return Node|null
+     * Find all nodes that are positioned _before_ the specified sibling and match the specified $filter
      */
-    public function findChildNodeConnectedThroughEdgeName(
-        NodeAggregateId $parentNodeAggregateId,
-        NodeName $edgeName
-    ): ?Node;
+    public function findPrecedingSiblingNodes(NodeAggregateId $siblingNodeAggregateId, Filter\FindPrecedingSiblingNodesFilter $filter): Nodes;
 
     /**
-     * @param NodeAggregateId $sibling
-     * @param FindSucceedingSiblingsFilter $filter
-     * @return Nodes
+     * Find a single child node by its name
+     *
+     * @return Node|null the node that is connected to its parent with the specified $edgeName, or NULL if no matching node exists or the parent node is not accessible
      */
-    public function findSucceedingSiblings(
-        NodeAggregateId $sibling,
-        Filter\FindSucceedingSiblingsFilter $filter
-    ): Nodes;
+    public function findChildNodeConnectedThroughEdgeName(NodeAggregateId $parentNodeAggregateId, NodeName $edgeName): ?Node;
 
     /**
-     * @param NodeAggregateId $sibling
-     * @param FindPrecedingSiblingsFilter $filter
-     * @return Nodes
+     *
```

Comment:
"recursively find all nodes underneath the entry node aggregate ID, which match the specified filter".